File: app/src/main/java/io/github/wulkanowy/ui/base/BaseContract.java
package io.github.wulkanowy.ui.base;

import android.support.annotation.StringRes;

public interface BaseContract {

    interface View {

        void onError(@StringRes int resId);

        void onError(String message);

        void onNoNetworkError();

        boolean isNetworkConnected();
    }

    interface Presenter<V extends View> {

        void onStart(V view);

        void onDestroy();
    }
}

File: app/src/main/java/io/github/wulkanowy/ui/base/BaseContract.kt
package io.github.wulkanowy.ui.base

interface BaseContract {

    interface View {

        fun showMessage(text: String)

        fun showNoNetworkMessage()
    }

    interface Presenter<T : BaseContract.View> {

        fun attachView(view: T)

        fun detachView()
    }
}
